src/timers.js

```diff
@@ -71,8 +71,9 @@ Promise.prototype.timeout = function (ms, message) {
     ms = +ms;
     var ret, parent;
 
+    var originalPromise = this;
     var handleWrapper = new HandleWrapper(setTimeout(function timeoutTimeout() {
-        if (ret.isPending()) {
+        if (originalPromise.isPending()) {
```

Comment:
If possible can you add a test to ensure that this code path is protected from future regressions?